### ISSUE TITLE
Bug (probably): unspecified types aren't errors

### ISSUE THREAD

**Issue by itamarst** [anweiss/cddl#106]
Consider the following schema:

```
reputation-object = {
        application: text
        reputons: [* reputon]
}
```

The `reputon` type is never defined, and yet the validation code seems perfectly happy with the following document:

```json
{"application": "blah", "reputons": [{"reputon": "xx"}
```

If this is invalid behavior, great, that can just be fixed.

But perhaps this is valid according to the RFC, and if so that's a very bad feature in a schema language, because a typo in a type name means validation suddenly doesn't happen. In which case perhaps a strict mode would be useful where all types must be explicitly specified? (But I really hope this isn't valid according to the RFC...).

**Comment by anweiss** [comment]
Indeed, I would interpret this as an _invalid_ JSON document against that schema because the `reputons` property exists and as such needs to be either an empty array or an array of `reputon`'s. Will investigate.

**Comment by anweiss** [comment]
This should be fixed as of [0.9.0-beta.1](https://github.com/anweiss/cddl/releases/tag/0.9.0-beta.1). Please re-open if this is still an issue.

**Comment by itamarst** [comment]
This fixed the problem for me when I tested latest version. Thank you!
